test/python/transpiler/test_inverse_cancellation.py

```diff
@@ -271,6 +271,74 @@ def test_cx_do_not_wrongly_cancel(self):
         self.assertIn("cx", gates_after)
         self.assertEqual(gates_after["cx"], 2)
 
+    def test_no_gates_to_cancel(self):
+        """Test when there are no gates to cancel."""
+        qc = QuantumCircuit(2)
+        qc.cx(0, 1)
+        qc.cx(1, 0)
+        inverse_pass = InverseCancellation([HGate()])
+        new_circ = inverse_pass(qc)
+        self.assertEqual(qc, new_circ)
+
+    def test_some_cancel_rules_to_cancel(self):
+        """Test when there are some gates to cancel."""
+        qc = QuantumCircuit(2)
+        qc.cx(0, 1)
+        qc.cx(1, 0)
+        qc.h(0)
+        qc.h(0)
+        inverse_pass = InverseCancellation([HGate(), CXGate(), CZGate()])
+        new_circ = inverse_pass(qc)
+        self.assertNotIn("h", new_circ.count_ops())
+
+    def test_no_inverse_pairs(self):
+        """Test when there are no inverse pairs to cancel."""
+        qc = QuantumCircuit(1)
+        qc.s(0)
+        qc.sdg(0)
+        inverse_pass = InverseCancellation([(TGate(), TdgGate())])
+        new_circ = inverse_pass(qc)
+        self.assertEqual(qc, new_circ)
+
+    def test_some_inverse_pairs(self):
+        """Test when there are some but not all inverse pairs to cancel."""
+        qc = QuantumCircuit(1)
+        qc.s(0)
+        qc.sdg(0)
+        qc.t(0)
+        qc.tdg(0)
+        inverse_pass = InverseCancellation([(TGate(), TdgGate())])
+        new_circ = inverse_pass(qc)
+        self.assertNotIn("t", new_circ.count_ops())
+        self.assertNotIn("tdg", new_circ.count_ops())
```

Comment:
Another another thought: it'd be good to include a test where the inverse pair are in the circuit back-to-front, e.g. `InverseCancellation([TGate(), TdgGate()]` on a circuit `tdg(0); t(0)`.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/11211/commits/f2f8be2ae2ca3fc7fb444132fa1f26ae666a9386